packages/openapi-typescript/src/transform/schema-object.ts

```diff
@@ -159,11 +159,30 @@ export function defaultSchemaObjectTransform(schemaObject: SchemaObject | Refere
     }
   }
 
-  // "type": "object" (explicit)
-  // "anyOf" / "allOf" (object type implied)
-
   // core type: properties + additionalProperties
   const coreType: string[] = [];
+
+  // discriminators
+  for (const k of ["oneOf", "allOf", "anyOf"] as ("oneOf" | "allOf" | "anyOf")[]) {
```

Comment:
Note: this was moved up **before property generation** due to the change in automatic `discriminator.propertyName` possibly adding keys to the object. We can’t detect if there may be a key conflict, but we can let the explicit properties override it.